models.py

```diff
@@ -110,66 +44,21 @@ def get_missions(self):
 		return Mission.query(Mission.runners == self.id).order(Mission.start).fetch(limit=None)
 	
 	def get_rank(self, semester=get_current_semester()):
-		paid = False
-		num_semesters_paid_to_date = 0
-		for semester_paid in self.semesters_paid:
-			if semester_paid <= semester:
-				paid = True
-			if semester_paid < semester:
-				num_semesters_paid_to_date += 1
-		
-		# Cadets cannot earn ranks
-		if not paid:
-			return 0
-		
-		rank = 1
-		
-		# Longevity
-		if num_semesters_paid_to_date >= 4:
-			rank += 1
-		
-		# Led weekly mission
-		if Mission.query(Mission.runners == self.id, Mission.type == 0, Mission.start < semester_date(semester)).count(limit=1) != 0:
-			rank += 1
-		
-		# Volunteered with special mission or committee
-		if self.committee_rank or Mission.query(Mission.runners == self.id, Mission.type == 1, Mission.start < semester_date(semester)).count(limit=1) != 0:
-			rank += 1
-		
-		# Merit ranks
-		if self.merit_rank1:
-			rank += 1
-		if self.merit_rank2:
-			rank += 1
-		
-		# Voted captain
-		if BridgeCrew.query(BridgeCrew.captain == self.id, BridgeCrew.start < semester_date(semester)).count(limit=1) != 0:
-			rank = 6
-		
-		# Alumni ranks
-		if not self.current_student:
-			if rank == 6:
-				# Captains become rear admirals
-				rank = 8
-			else:
-				# Non-captains become commodores
-				rank = 7
-		
-		# Was an admiral (advisor)
-		if BridgeCrew.query(BridgeCrew.admiral == self.id, BridgeCrew.start < semester_date(semester)).count(limit=1) != 0:
-			rank = 10
-		
-		return rank
+		from ranks import rank
```

Comment:
Why import here instead of with the other imports at the top?

Reply:
It solves an error that prevents importing classes from models in ranks.py. If they are at the top of the file it tries to import from ranks when ranks runs an import from models (of classes that are not Member) and the app breaks.